### DISCUSSION TITLE
Events' sticky player on the right side

### DISCUSSION THREAD

**Discussion by decimoseptimo** [blakeblackshear/frigate#6107] [Ideas]
![Untitled](https://user-images.githubusercontent.com/16873203/232398172-cf07a3c9-a362-4ed3-b3c2-a8a95bdfed65.png)

Wouldn't it make sense to have the player fixed on the right side, while the event cards be unlimited scrolled in the middle, for desktops?
And how relevant is still this events view considering there's an experimental ui in the works?

**Comment by NickM-27** [comment]
That's an interesting approach, would mean the first video needs to be selected by default or there would be a lot of empty space. 

And the experimental UI hasn't been changed since 0.11, it doesn't support the latest live views. There will most likely be a different solution that is implemented in an upcoming release

**Reply by decimoseptimo** [reply]
> That's an interesting approach, would mean the first video needs to be selected by default or there would be a lot of empty space.

Perhaps a video player not playing, or the default birdseye image.

What about [native scrolling](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scrollbars) vs a [light dependency](https://grsmto.github.io/simplebar/examples/)?

**Comment by decimoseptimo** [comment]
I just reviewed the 2 player options in the works

- https://github.com/blakeblackshear/frigate/pull/1155#issuecomment-854948555
ui seems old fashioned

- https://github.com/blakeblackshear/frigate/issues/2230
ui looks modern

Let's say the second one is chosen over the first one, following that design pattern https://youtu.be/DdchTdYo8KA?t=751
we'll probably still need a 'full history' view, like the revamped events view i propose here. Where in mobile, videos would open up fullscreen, instead of the current accordeon behaviour.

**Reply by NickM-27** [reply]
The first is the current and the second is hidden behind an experimental ui flag. There's other options and we've already discussed what we see as a good way forward. It will be coming in a future version. 

In any case, if you have something specific you want to contribute, feel free to make a PR

**Reply by decimoseptimo** [reply]
May you point me to those options or discussions. Updating the events view seems a significant feat, i think it's worth having a clear definition of what's needed. Here I'm just pouring some ideas, but i don't know what the vision is.

**Reply by NickM-27** [reply]
it would actually be an update to the recordings view, not the events view (what I am referring too, anyway). and the discussions weren't here on github.

ideas are welcome, but at the end of the day it is going to be up to the person implementing the change and the reviewers to determine what things will look like.

**Reply by decimoseptimo** [reply]
I dont know what a recordings view is. Perhaps the template for `/events` url?

> ideas are welcome, but at the end of the day it is going to be up to the person implementing the change and the reviewers to determine what things will look like.

In my opinion/experience ideas > design > code.  This is the spark of an idea, i included some design, but i dont know how it plays out in your vision/roadmap. Aditionally the 2 coming up aproaches adds to the confusion. I dont consider i'm at the point to start codding.
Ideally i'd like to see the designs to see where and if it makes sense to contribute to.

**Reply by NickM-27** [reply]
> I dont know what a recordings view is. Perhaps the template for `/events` url?

The recording view is  https://github.com/blakeblackshear/frigate/pull/1155#issuecomment-854948555 which is available in the current version of frigate at `http://frigate_ip:5000/recording/camera_name`

it now occurs to me that this may be a big point of confusion here, since this recordings view is not upcoming it has existed for many versions. Frigate has an event view and a 24/7 recordings view available today.

**Reply by decimoseptimo** [reply]
Yes, `/recordings` is not on the main menu, so I overlooked it. Here were my findings:

- It's buggy, i've seen page reloads after skipping several videos downwards, i might be when a new "hour set" is played.
- nav on top of video suits better for fullscreen (particularly looks bad on mobile non-fs with so very little space). I think an independent nav like this i proppose is better for when not in fullscreen as it lets you browse while playing.
- It showcases events more than it does hours. I have a little harder time undestanding what it did.

Unless i'm missing something, i don't see the need for having to 2 different player ui's that do mostly the same.

Perhaps have an independent nav when not FS, and nav on top of video when FS, for desktop?

**Reply by blakeblackshear** [reply]
The UI is mostly the way it is because of the evolution of features in frigate, not because it was intentionally designed to have recordings and events separate.

The long term vision is to have something more similar to the unifi protect and nest aware interface. I like vertical timeline scrolling on mobile and horizontal scrolling on desktop. When scrolling to the end, you should flip to a live view. One interface for live and historical footage. We have already been making backend changes to support that future direction. The backend is moving in a direction where it stores data about what happened in history agnostic of the video. For example, we already have motion and object data available on every recording segment. This would support a UI that overlays motion activity and object info on a timeline. 

One UX concept I want to incorporate is something similar to the minimap in vs code for large files. We should be able to provide context for an entire day and simply tap/drag to navigate to that part of the timeline. Then you can scroll to refine what you want to watch. I also like the youtube feature of dragging up on the timeline for more precise seeking.

This just hasn't hit the priority list because it's not really a differentiator for frigate. We are still focused on improving the accuracy of detection and object tracking to the point that it can be used more reliably for accurate automations and security focused alerts. I want frigate to be best in class for that task. The UI updates will come after that.

If you are interested in contributing to the UI, I would suggest creating some wire frames to propose before spending the time.

**Reply by decimoseptimo** [reply]
https://www.figma.com/file/rFz69Zjzy2d6MMEHOMj6jM/Frigate?node-id=0%3A1&t=AENEUAWXf05tzKIb-1
I made this, and left some comments

**Reply by Majestic7979** [reply]
@blakeblackshear 

I will eat dog poop tomorrow if you make the UI a priority.

I know, your project = your way. And there's absolutely nothing wrong with wanting to make your differentiator "best in class detection". But trust me - please read this to understand the living context of anyone living outside of a suburb in North America (https://github.com/blakeblackshear/frigate/issues/8183#issuecomment-1766845726)... the UI is 1000000000% important. I have just felt confident enough ditching my Nest Hello to let Frigate handle *ALL* of my recordings. But this will introduce a level of pain that you simply cannot imagine unless you're wearing my (very clean and nice smelling) shoes. Detections are fine but I still have to sift through 24/7 footage to find what I need when there's hundreds of people walking across the front door every single day or night.

Many of us in Europe live in busy roads, and the only way to find out what happened is literally by scrolling through a timeline until you find what you need. Currently in Frigate this is a nightmare. I am a Neighborhood Watch Coordinator in my area, and I get law enforcement requests all the time. I can find stuff on Nest Aware very quickly because of their seamless timeline with those little dots to show events. But on Frigate it's a horrendous task of manually clicking events and hours and days...

If you already have all the data points, PLEASE I BEG YOU ON BEHALF OF ANYONE WHO'S NOT IN A MCMANSION IN A USA SUBURB: MAKE THE UI UPGRADE HAPPEN. Get it out of the way, and then focus on the "mission-critical" stuff no matter how hard it is. For anyone to adopt your solution it's gonna have to be user-friendly first and foremost. Nobody but a bunch of nerds like us will care unless it looks like what they're already used to. I have a bunch of clients that would immediately say yes to Frigate if it offered a better way to browse through security footage, but if I showed it to them now they'd throw the tablet on my face.

**Reply by decimoseptimo** [reply]
Each to their own. I'm having a hard time detecting night colorvu stuff. So as much as i like UI/UX, i can see why detection is a priority.

**Reply by Zandiexoder** [reply]
100% agree, a good quality UI update should last Frigate years of updates before a change is needed.

**Reply by kkoshelev** [reply]
Speaking from personal point of view. I don't care about detection that much, but existing UI is a big deal breaker for the adoption. I would go and purchase 5 more IP cameras today and climb the attic to replace the existing Nest cameras if UI had a proper timeline and mobile browser support.

**Reply by NickM-27** [reply]
To be clear the UI is currently being rewritten from the ground up, there is no reason to continue to advocate for a new UI

**Reply by Majestic7979** [reply]
I'm very interested in seeing your ideas... can you share the mockups?

**Reply by NickM-27** [reply]
The UI rewrite is already very much in progress (as can be seen by the activity in dev). It is not currently in a place where we are looking for feedback and is rapidly changing so in general not recommended for users to run

**Reply by NickM-27** [reply]
for those who missed it, the new UI has been announced https://github.com/blakeblackshear/frigate/discussions/11136

**Comment by Majestic7979** [comment]
> The UI rewrite is already very much in progress (as can be seen by the activity in dev). It is not currently in a place where we are looking for feedback and is rapidly changing so in general not recommended for users to run

Please don't get me wrong I don't mean to offer any input whatsoever, just curious... Not looking to run it or anything like that. Pinky promise. Only wanna take a look. A few screenshots would kill the curiosity... Just trying to see where it's heading. I promise to bite my tongue.

**Reply by NickM-27** [reply]
see the announcement https://github.com/blakeblackshear/frigate/discussions/11136
